src/main/java/org/spongepowered/api/GameRegistry.java

```diff
@@ -96,11 +113,29 @@
      * @param id The case insensitive string id of the dummy type
      * @param <T> The type of dummy type
      * @return The found dummy type, if available
+     * @deprecated use {@link #getType(Class, CatalogKey)}
      * @see CatalogType
      */
+    @Deprecated
```

Comment:
I disagree with the deprecation. Sure we highly encourage usage of a CatalogKey but it shouldn't be a requirement: if someone passes a string that can't be mapped to anything of the type then they get back the empty optional.

Reply:
Agreed with Zidane, it should be at the base level lowest level allowance to use Strings. We want to avoid the little issues we've had with usability of `DataQuery`s.

Reply:
We're going to become like `ResourceLocation` in Forget-me-not, then. Forge is stuck with `String` ids everywhere.
`String` based ids are just plain silly. It results in us needing to guess what namespace to prepend to the string they provide, it has no documentation that can be seen during creation, it requires a bunch of spots throughout the implementation (which we already have a huge mess of, which this PR set will fix) changing the case to lowercase, etc.

From the Oracle documentation:
> When you design an API, carefully consider whether it supersedes an old API. If it does, and you wish to encourage developers (users of the API) to migrate to the new API, then deprecate the old API. Valid reasons to deprecate an API include:
>
> - It is insecure, buggy, or highly inefficient
> - It is going away in a future release
>- It encourages bad coding practices
>
>Deprecation is a reasonable choice in all these cases because it preserves "backward compatibility" while encouraging developers to change to the new API. Also, the deprecation comments help developers decide when to move to the new API, and so should briefly mention the technical reasons for deprecation.

It would be stupid to keep the `String`-based methods non-deprecated.

Reply:
After discussion with @kashike he convinced me to allow the deprecation here with a compromise that it'll be a bit before we remove it. I do agree with him on the fact that the Forge ecosystem cannot move past the usage of the strings for the keys because it was never deprecated correctly and now impossible to remove.